eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringMongoReader.java

```diff
@@ -106,13 +161,15 @@ public void initializeReader() {
     private FindIterable<Document> getSubmittedVariants(Bson query, Class<?> entityClass) {
         return mongoTemplate.getCollection(mongoTemplate.getCollectionName(entityClass))
                             .find(query)
+                            .sort(Sorts.ascending(ID_FIELD))
```

Comment:
Does adding a limit function with the chunkSize cause any issues here? If not, I will recommend adding it because I am not sure if Spring does that internally.

Reply:
I think setting the cursor's batchSize should be enough, right?  Otherwise we'd have to call `initializeReader` for every chunk...